examples/http/onvif_camera.js

```diff
@@ -0,0 +1,224 @@
+    port: 554
+};
+
+// Table for capturing status
+var table = D.createTable(
+    "Video Streaming Status",
+    [
+        { label: "Streaming status" }
+    ]
+);
+
+/**
+ * Retrieves the profiles from the ONVIF camera.
+ * @returns A promise that resolves with an array of profile tokens.
+ */
+function getProfiles() {
+    var d = D.q.defer();
+    var config = {
+        url: url,
+        username: username,
+        password: password,
+        auth: "basic",
+        body: '<?xml version="1.0" encoding="UTF-8"?>\n<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:trt="http://www.onvif.org/ver10/media/wsdl">\n<soap:Body>\n<trt:GetProfiles/>\n</soap:Body>\n</soap:Envelope>' 
+    };
+    D.device.http.post(config, function(error, response, body){
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (!response) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode == 400) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 200) {
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else {
+
+            var $ = D.htmlParse(body);
+            var profiles = $("trt\\:Profiles"); 
+            var profileTokens = [];
+            profiles.each(function(index, element) {
+                var token = $(element).attr("token");
+                profileTokens.push(token);
+            });
+            d.resolve(profileTokens);
+        }
+    });
+    return d.promise;
+}
+
+/**
+ * Retrieves the stream URIs for each profile token
+ * @param {Array} profileTokens Array of profile tokens
+ * @returns A promise that resolves with an array of stream URIs
+ */
+function getStreamURIs(profileTokens){
+    var xmlPayload = '<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:trt="http://www.onvif.org/ver10/media/wsdl"><soap:Body>';
+
+    var promises = [];
+   
+    profileTokens.forEach(function(profileToken) {
+        var d = D.q.defer();
+        var streamUris = xmlPayload + '<trt:GetStreamUri>' +
+                                      '<trt:StreamSetup>' +
+                                      '<trt:Stream xmlns:trt="http://www.onvif.org/ver10/media/wsdl">RTP-Unicast</trt:Stream>' +
+                                      '<trt:Transport xmlns:trt="http://www.onvif.org/ver10/media/wsdl">' +
+                                      '<trt:Protocol>RTSP</trt:Protocol>' +
+                                      '</trt:Transport>' +
+                                      '</trt:StreamSetup>' +
+                                      '<trt:ProfileToken>' + profileToken + '</trt:ProfileToken>' +
+                                      '</trt:GetStreamUri>' +
+                                      '</soap:Body>' +
+                                      '</soap:Envelope>';
+
+        var config = {
```

Comment:
onvif port can (and usually is) different than the default http port. Please add as parameter with default 80